### PR TITLE
fix: fixes for round robin reassignment

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

Some fixes for the new manual round robin reassignment https://github.com/calcom/cal.com/pull/17215

- email was sent to wrong person (fixed host received cancelled email instead of the previous RR host)
- email and calendar event showed wrong  attendees and team members 
- scheduled email showed content of cancellation email

### CODE REVIEW

File: packages/features/ee/round-robin/roundRobinReassignment.ts [hunk 1]
```diff
@@ -151,17 +151,6 @@ export const roundRobinReassignment = async ({
...
-  // Assume the RR host was labelled as a team member
-  if (reassignedRRHost.email !== organizer.email) {
-    teamMembers.push({
-      id: reassignedRRHost.id,
-      email: reassignedRRHost.email,
-      name: reassignedRRHost.name || "",
-      timeZone: reassignedRRHost.timeZone,
-      language: { translate: reassignedRRHostT, locale: reassignedRRHost.locale ?? "en" },
-    });
-  }
```
Comment (CarinaWolli) [inline packages/features/ee/round-robin/roundRobinReassignment.ts:163]: The new host was shown double in the email. @sean-brydon can you make sure this doesn't break anything?

File: packages/features/ee/round-robin/roundRobinManualReassignment.ts [hunk 2]
```diff
@@ -166,24 +184,38 @@ export const roundRobinManualReassignment = async ({
...
+    attendeePromises.push(
+      getTranslation(attendee.locale ?? "en", "common").then((tAttendee) => ({
+        email: attendee.email,
+        name: attendee.name,
+        timeZone: attendee.timeZone,
+        language: { translate: tAttendee, locale: attendee.locale ?? "en" },
+        phoneNumber: attendee.phoneNumber || undefined,
+      }))
+    );
+  }
```
Comment (CarinaWolli) [inline packages/features/ee/round-robin/roundRobinManualReassignment.ts:218]: same logic as we use in `roundRobinReassignment.ts`

File: packages/features/ee/round-robin/roundRobinManualReassignment.ts [hunk 3]
```diff
@@ -242,8 +274,10 @@ export const roundRobinManualReassignment = async ({
     newReferencesToCreate,
   });
 
+  const { cancellationReason, ...evtWithoutCancellationReason } = evt;
+
   // Send emails
-  await sendRoundRobinScheduledEmailsAndSMS(evt, [
+  await sendRoundRobinScheduledEmailsAndSMS(evtWithoutCancellationReason, [
```
Comment (CarinaWolli) [inline packages/features/ee/round-robin/roundRobinManualReassignment.ts:280]: if `cancellationReason` is set, email scheduled email looks like that: 
Screenshot 2024-10-22 at 12 51 10 AM

File: packages/features/ee/round-robin/roundRobinManualReassignment.ts [hunk 4]
```diff
@@ -266,11 +300,11 @@ export const roundRobinManualReassignment = async ({
     cancelledEvt,
     [
       {
-        ...originalOrganizer,
-        name: originalOrganizer.name || "",
-        username: originalOrganizer.username || "",
-        timeFormat: getTimeFormatStringFromUserTimeFormat(originalOrganizer.timeFormat),
-        language: { translate: originalOrganizerT, locale: originalOrganizer.locale || "en" },
+        ...previousRRHost,
```
Comment (CarinaWolli) [inline packages/features/ee/round-robin/roundRobinManualReassignment.ts:303]: The `previousRRHost` needs to get the cancellation emails. The orginalOriganizer could be a fixed host

**Review by sean-brydon — APPROVED** [review]
LGTM - Added a test for the case of sending the cancellation email

**Pull request merged**